### PR TITLE
Ensure renamePending gets reset when operation is complete

### PR DESCRIPTION
**Pull request by Colengms**
These are the changes I suggested on https://github.com/microsoft/vscode-cpptools/pull/4342

This does not address the condition we've discussed needing to be restored in cancelReferences.  I think that may not actually be needed.  As long as pendingRename is getting properly reset as the (last) rename operation is completed successfully, we shouldn't see unwanted calls to cancelReferences() after a completed rename.

### CODE REVIEW

**Comment by michelleangela** [comment]
Note: added this changeset to #4342 and it doesn't handle the case:

1. Do a rename with unconfirmed references
2. edit file to cause rename to cancel
Result: does not cancel rename

Investigating still...

**Review by sean-mcmanus — APPROVED** [review]

File: Extension/src/LanguageServer/references.ts [hunk 1]
```diff
@@ -343,7 +344,20 @@ export class ReferencesManager {
...
+            if (this.currentUpdateProgressTimer) {
+                clearInterval(this.currentUpdateProgressTimer);
+                this.currentUpdateProgressResolve();
+                this.currentUpdateProgressResolve = null;
+                this.currentUpdateProgressTimer = null;
+            }
+            this.client.setReferencesCommandMode(ReferencesCommandMode.None);
+        }
+
+        if (currentReferenceCommandMode === ReferencesCommandMode.Rename) {
```
Comment (sean-mcmanus) [inline Extension/src/LanguageServer/references.ts:360]: Can this be an "else if"?

Comment (sean-mcmanus) [reply Extension/src/LanguageServer/references.ts:360]: Looks like it can't...

Comment (Colengms) [reply Extension/src/LanguageServer/references.ts:360]: Are you suggesting if isFinished, don't do the rest?  I think the rest is still needed even if isFinished.

**Pull request merged**